tests/test_utils_gz.py

```diff
@@ -25,5 +25,10 @@ def test_gunzip_no_gzip_file_raises(self):
 
     def test_gunzip_truncated_short(self):
         with open(join(SAMPLEDIR, 'truncated-crc-error-short.gz'), 'rb') as f:
-            text = gunzip(f.read())
+            text = gunzip(f.read()) 
             assert text.endswith(b'</html>')
+
+    def test_gunzip_illegal_eof(self):
+        with open(join(SAMPLEDIR, 'unexpected-eof.gz'), 'rb') as f:
+            text = gunzip(f.read())
+            self.assertEqual(text.count(b'document'), 1)
```

Comment:
I did create a file in the sense that I saved the gzipped response sent by the server to a file, however I do agree the test in itself isn't solid enough. Maybe I can fetch the un-compressed text using the `requests` library, save it to a file in `SAMPLEDIR` and compare the body of that to the output of gunzip?